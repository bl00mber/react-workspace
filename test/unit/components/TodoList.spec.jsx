import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Todo from '@/components/Todo';
import TodoList from '@/components/TodoList';

describe('TodoList', () => {
  test('should match enzyme shallow snapshot', () => {
    const props = {
      todos: [
        { id: 1, completed: false, text: 'todo 1' },
        { id: 2, completed: true, text: 'todo 2' },
      ],
      onTodoClick: jest.fn(),
    };
    const wrapper = shallow(<TodoList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should call onTodoClick clicking on todo element', () => {
    const props = {
      todos: [
        { id: 1, completed: false, text: 'todo 1' },
        { id: 2, completed: true, text: 'todo 2' },
      ],
      onTodoClick: jest.fn(),
    };
    const wrapper = shallow(<TodoList {...props} />);
    wrapper.find(Todo).at(0).simulate('click');
    expect(props.onTodoClick).toHaveBeenCalledWith(1);
    wrapper.find(Todo).at(1).simulate('click');
    expect(props.onTodoClick).toHaveBeenCalledWith(2);
  });
});
