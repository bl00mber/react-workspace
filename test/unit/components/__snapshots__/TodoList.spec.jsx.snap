// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should match enzyme shallow snapshot 1`] = `
<ul>
  <Todo
    completed={false}
    id={1}
    key="1"
    onClick={[Function]}
    text="todo 1"
  />
  <Todo
    completed={true}
    id={2}
    key="2"
    onClick={[Function]}
    text="todo 2"
  />
</ul>
`;
